Updated directory structure for k8s manifest files (#201)
prometheus/test-infra: README.md

It runs with [Prow CI](https://github.com/kubernetes/test-infra/blob/master/prow/) on a [Google Kubernetes Engine Cluster](https://cloud.google.com/kubernetes-engine/).
It is designed to support adding more k8s providers.

## Overview of the manifest files
The `/manifest` directory contains all the kubernetes manifest files.

> - `cluster.yaml` : This is used to create the GKE cluster.

Reviewer: I think we can move this file inside `cluster-infra/` to simplify the Readme and deployment even more.

Reviewer: can you move the cluster.yaml inside the cluste-infra folder?

Author: oh sorry i missed this one. updated.

> - `cluster-wide/` : These are the persistent k8s resources.

Reviewer: How about renaming the folder to `cluster-infra`?
```suggestion
- `cluster-infra/` : These are the persistent cluster infrastructure resources.
```

- `prombench/` : These resources are created and destoryed for each prombench test.

> - `prow/` : Resources for deploying [prow](https://github.com/kubernetes/test-infra/tree/master/prow/), which helps in triggering the tests from GitHub.

Reviewer: ```suggestion
- `prow/` : Resources for deploying [prow](https://github.com/kubernetes/test-infra/tree/master/prow/), which is used to trigger tests from GitHub comments.
```

export AUTH_FILE=<path to service-account.json>

./prombench gke cluster create -a $AUTH_FILE -v PROJECT_ID:$PROJECT_ID \

>     -v ZONE:$ZONE -v CLUSTER_NAME:$CLUSTER_NAME -f manifests/cluster-infra/0_cluster.yaml

Reviewer: now with this change the Readme can also be simplified.

Author: oh i just noticed the commands are different for deploying the `cluster.yaml` and the other k8s objects. as `cluster.yaml` is for gke. should we revert the change?

Reviewer: can't we deploy these in a single step and adjust the Readme?

Reviewer: aaah I see now, these are different commands - cluster create vs resource apply. so yes need to revert the change.

Author: I updated it now.

```
./prombench gke resource apply -a $AUTH_FILE -v PROJECT_ID:$PROJECT_ID -v ZONE:$ZONE \
    -v CLUSTER_NAME:$CLUSTER_NAME -v GCLOUD_SERVICEACCOUNT_CLIENT_EMAIL:$GCLOUD_SERVICEACCOUNT_CLIENT_EMAIL \
    -f manifests/cluster-wide
```

> This will be good time to set the A record for `<DOMAIN_NAME>` to the nginx-ingress-controller IP address. See [Deploy Prometheus-Meta & Grafana](#deploy-prometheus-meta--grafana-1).

Reviewer: instead of this can just move 	this section here
https://github.com/prometheus/prombench/pull/201/files#diff-04c6e90faac2675aa89e2176d2eec7d8L147

btw is using a domain optional? If someone wants to run a quick test without setting a domain name, is this possible?

Author: The services will be available in the ingress IP but since `--web.external-url` flag is used for prometheus and the domain name is also used in the grafana data source, some parts will not function properly without setting up the domain pointing to the ingress IP.

not having a domain point to the ingress IP will not break the deployment but all functionalities will not function as expected.

export GRAFANA_ADMIN_PASSWORD=password

> export DOMAIN_NAME=prombench.prometheus.io

Reviewer: `export DOMAIN_NAME=prombench.prometheus.io // Can be set to any other custom domain.`

```

- Deploy the [nginx-ingress-controller](https://github.com/kubernetes/ingress-nginx), Prometheus-Meta & Grafana.
```
./prombench gke resource apply -a $AUTH_FILE -v PROJECT_ID:$PROJECT_ID -v ZONE:$ZONE \
    -v CLUSTER_NAME:$CLUSTER_NAME -v DOMAIN_NAME:$DOMAIN_NAME \
    -v GRAFANA_ADMIN_PASSWORD:$GRAFANA_ADMIN_PASSWORD \
    -v GCLOUD_SERVICEACCOUNT_CLIENT_EMAIL:$GCLOUD_SERVICEACCOUNT_CLIENT_EMAIL \
    -f manifests/cluster-wide
```

> - Get the `nginx-ingress-controller IP address` from the logs of this command, alternatively you can get it from the GKE/Services tab.

Reviewer: ```suggestion
- The output will show the ingress IP which will be used to point the domain name to. Alternatively you can see it from the GKE/Services tab.
```

> - Set the `A record` for `<DOMAIN_NAME>` to point to `nginx-ingress-controller IP address`, it can be any domain/subdomain.

Reviewer: ```suggestion
- Set the `A record` for `<DOMAIN_NAME>` to point to `nginx-ingress-controller` IP address.
```

- The services will be accessible at:
  * Grafana :: `http://<DOMAIN_NAME>/grafana`
  * Prometheus ::  `http://<DOMAIN_NAME>/prometheus-meta`

>   * Prow dashboard :: `http://<DOMAIN_NAME>` (once [deploying prow](#deploy-prow) is completed)

Reviewer: ```suggestion
```

### Deploy Prow
> This is used to monitor GitHub comments and starts new tests.

---

- Follow [Setting GitHub API and webhook](#setting-up-github-api-and-webhook)

- Add all required tokens as k8s secrets.
  * hmac is used when verifying requests from GitHub.
  * oauth is used when sending requests to the GitHub api.
  * gke auth is used when scaling up and down the cluster.
```
./prombench gke resource apply -a $AUTH_FILE -v ZONE:$ZONE \
    -v CLUSTER_NAME:$CLUSTER_NAME -v PROJECT_ID:$PROJECT_ID \
    -v HMAC_TOKEN="$(printf $HMAC_TOKEN | base64 -w 0)" \
    -v OAUTH_TOKEN="$(printf $OAUTH_TOKEN | base64 -w 0)" \
    -v GKE_AUTH="$(cat $AUTH_FILE | base64 -w 0)" \
    -f manifests/prow/secrets.yaml
```

- Deploy all internal prow components

```
export GITHUB_ORG=prometheus
export GITHUB_REPO=prometheus

./prombench gke resource apply -a $AUTH_FILE -v PROJECT_ID:$PROJECT_ID \
    -v ZONE:$ZONE -v CLUSTER_NAME:$CLUSTER_NAME -v DOMAIN_NAME:$DOMAIN_NAME \
    -v GITHUB_ORG:$GITHUB_ORG -v GITHUB_REPO:$GITHUB_REPO \
    -f manifests/prow/components

> ```

Reviewer: can add here 

Prow can be accessed at :: `http://<DOMAIN_NAME>`

```
./prombench gke nodepool create -a $AUTH_FILE \
    -v ZONE:$ZONE -v PROJECT_ID:$PROJECT_ID -v CLUSTER_NAME:$CLUSTER_NAME \
    -v PR_NUMBER:$PR_NUMBER -f manifests/prombench/nodepools.yaml
```

- Deploy the k8s objects
```
./prombench gke resource apply -a $AUTH_FILE \
    -v ZONE:$ZONE -v PROJECT_ID:$PROJECT_ID -v CLUSTER_NAME:$CLUSTER_NAME \
    -v PR_NUMBER:$PR_NUMBER -v RELEASE:$RELEASE -v DOMAIN_NAME:$DOMAIN_NAME \
    -f manifests/prombench/benchmark
```

> ### Setting up GitHub API and webhook

Reviewer: ```suggestion
### Setting up GitHub API and webhook to trigger tests from comments.
```
